greenpithumb/pi_io.py

```diff
@@ -0,0 +1,43 @@
+#import RPi.GPIO as GPIO
+
+# TODO(mtlynch): Move these pin constants to a configuration file so they're not
+# hardcoded in.
+
+# Raspberry Pi pin to which each component is connected
+PIN_PUMP = 0
+
+
+class IO(object):
+    """Wrapper for input and output on a Raspberry Pi board.
+
+    This wraps the board and allows the caller to read or send signals to
+    the Raspberry Pi's pins.
+    """
+
+    def __init__(self, gpio):
+        """Creates a new input/output wrapper.
+
+        Args:
+            gpio: Raspberry Pi GPIO module.
+        """
+        self._GPIO = gpio
+
+    def turn_pin_on(self, pin):
+        """Turns on a Raspberry Pi GPIO pin.
+
+        Args:
+            pin: Index of Raspberry Pi pin to turn on
+        """
+        self._GPIO.setup(pin, self._GPIO.OUT)
+        self._GPIO.output(pin, 1)
+        self._GPIO.cleanup(pin)
```

Comment:
Calling `GPIO.cleanup` turns off all the GPIO pins and I think prevents us from calling `GPIO` for the rest of the program, so we only want to do that at the end.

Probably the best way to address is to have a `close` method that calls `cleanup`. We can clean it up later to use context managers (or something like [contextlib.closing](https://docs.python.org/2.7/library/contextlib.html)) but just having a `close` method and documenting that the caller has to call it when done should be enough for now.

Also, the caller should only ever create one of these instances, but we'll just document that and look at ways to clean it up later.